modules/programs/neovim.nix

```diff
@@ -217,6 +219,38 @@ in {
           This option is mutually exclusive with <varname>configure</varname>.
         '';
       };
+
+      coc = {
+        enable = mkEnableOption "Coc";
+
+        settings = mkOption {
+          type = jsonFormat.type;
+          default = { };
+          example = literalExample ''
+            {
+              "suggest.noselect": true,
+              "suggest.enablePreview": true,
+              "suggest.enablePreselect":false,
+              "suggest.disableKind": true,
+              "languageserver": {
+                "haskell": {
+                  "command": "haskell-language-server-wrapper",
+                  "args": ["--lsp"],
+                  "rootPatterns": ["*.cabal", "stack.yaml", "cabal.project", "package.yaml", "hie.yaml"],
+                  "filetypes": ["haskell", "lhaskell"]
+                }
+              }
```

Comment:
Can you change the code to use the nix syntax since we are no longer using `types.lines`

Reply:
Changed